test/test_little_dict.jl

```diff
@@ -0,0 +1,516 @@
+        @test FF1 <: FrozenLittleDict{Int, Int}
+        @test !(FF1 <: UnfrozenLittleDict{<:Any, <:Any})
+        @test !(FF1 <: UnfrozenLittleDict)
+        @test !(FF1 <: UnfrozenLittleDict{Int, Int})
+
+
+        UU1 = LittleDict{Int,Int,Vector{Int},Vector{Int}}
+        @test !(UU1 <: FrozenLittleDict{<:Any, <:Any})
+        @test !(UU1 <: FrozenLittleDict)
+        @test !(UU1 <: FrozenLittleDict{Int, Int})
+        @test (UU1 <: UnfrozenLittleDict{<:Any, <:Any})
+        @test (UU1 <: UnfrozenLittleDict)
+        @test (UU1 <: UnfrozenLittleDict{Int, Int})
+
+
+        FU1 = LittleDict{Int,Int,NTuple{10, Int},Vector{Int}}
+        @test !(FU1 <: FrozenLittleDict{<:Any, <:Any})
+        @test !(FU1 <: FrozenLittleDict)
+        @test !(FU1 <: FrozenLittleDict{Int, Int})
+        @test !(FU1 <: UnfrozenLittleDict{<:Any, <:Any})
+        @test !(FU1 <: UnfrozenLittleDict)
+        @test !(FU1 <: UnfrozenLittleDict{Int, Int})
+
+        UF1 = LittleDict{Int,Int,Vector{Int},NTuple{10,Int}}
+        @test !(UF1 <: FrozenLittleDict{<:Any, <:Any})
+        @test !(UF1 <: FrozenLittleDict)
+        @test !(UF1 <: FrozenLittleDict{Int, Int})
+        @test !(UF1 <: UnfrozenLittleDict{<:Any, <:Any})
+        @test !(UF1 <: UnfrozenLittleDict)
+        @test !(UF1 <: UnfrozenLittleDict{Int, Int})
+    end
+
+    @testset "Constructors" begin
+        @test isa(@inferred(LittleDict()), LittleDict{Any,Any})
+        @test isa(@inferred(LittleDict([(1,2.0)])), LittleDict{Int,Float64})
+
+        @test isa(@inferred(LittleDict([("a",1),("b",2)])), LittleDict{String,Int})
+        @test isa(@inferred(LittleDict(Pair(1, 1.0))), LittleDict{Int,Float64})
+        @test isa(@inferred(LittleDict(Pair(1, 1.0), Pair(2, 2.0))),
+        LittleDict{Int,Float64})
+
+        @test isa(@inferred(LittleDict{Int,Float64}(2=>2.0, 3=>3.0)),
+            LittleDict{Int,Float64})
+        @test isa(@inferred(LittleDict{Int,Float64}(Pair(1, 1), Pair(2, 2))), LittleDict{Int,Float64})
+        @test isa(@inferred(LittleDict(Pair(1, 1.0), Pair(2, 2.0), Pair(3, 3.0))), LittleDict{Int,Float64})
+        @test LittleDict(()) == LittleDict{Any,Any}()
+
+        @test isa(@inferred(LittleDict([Pair(1, 1.0), Pair(2, 2.0)])), LittleDict{Int,Float64})
+        @test_throws ArgumentError LittleDict([1,2,3,4])
+
+        iter = Iterators.filter(x->x.first>1, [Pair(1, 1.0), Pair(2, 2.0), Pair(3, 3.0)])
+        @test @inferred(LittleDict(iter)) == LittleDict{Int,Float64}(2=>2.0, 3=>3.0)
+        
+        iter = Iterators.drop(1:10, 1)
+        @test_throws ArgumentError LittleDict(iter)
+
+        k_iter = Iterators.filter(x->x>1, [1,2,3,4])
+        v_iter = Iterators.filter(x->x>1, [1.0,2.0,3.0,4.0])
+        @test @inferred(LittleDict(k_iter, v_iter)) isa
+            LittleDict{Int,Float64, Vector{Int}, Vector{Float64}}
+
+        @test @inferred(LittleDict{Int, Char}(rand(1:100,20), rand('a':'z', 20))) isa
+            LittleDict{Int64,Char,Array{Int64,1},Array{Char,1}}
```

Comment:
I think `Int64` needs to be `Int` here for tests to pass on 32-bit systems.

Reply:
```suggestion
            LittleDict{Int,Char,Array{Int,1},Array{Char,1}}
```